examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -119,7 +120,11 @@ CHIP_ERROR PairingCommand::Pair(NodeId remoteId, PeerAddress address)
     else
     {
         auto commissioningParams = GetCommissioningParameters();
-        err                      = CurrentCommissioner().PairDevice(remoteId, params, commissioningParams);
+        if (mPairingMode == PairingMode::Ble || mPairingMode == PairingMode::SoftAP)
```

Comment:
How will this work in a real commissioner, which is given a code and does discovery, not told "pair over SoftAP"?

The commissioner will pass that code into CHIPDeviceController to do the PASE setup and commissioning ... and then what?

Reply:
For chip-tool, we will use `pairing code-wifi` to tell the commissioner that it needs to setup the commissionee's operational network. This command will also pass the WiFi credentials into CHIPDeviceController and make commissionee join the operational WiFi network after sending NOC to commissionee.

Reply:
I understand what happens for chip-tool.  I am asking about "a real commissioner": a thing that gets to scan a QR code and then needs to commission the device.  How is this API supposed to work for those?

Reply:
Is it OK that we always pass the WiFi credentials and setup the commissionee's network for the code pairing? If the commissionee has already been connected to the passed WiFi network(onnetwork pairing), it will return status OK directly, just like #23251. And if the commissionee is not connected to any network (ble pairing or softAP pairing), it will join the passed WiFi network.

Reply:
> For a commissioner which scans a QR code, there is Discovery Capabilities Bitmask with a bit for SoftAP commissioning. If this bit is set to 1, the commissioner will discover the device via SoftAP DNS-SD and do network setup after sending the NOC.

Yes, but it might still discover the commissioner on the operational network or not...

Put another way: why do we need this extra API to handle cases (chip-tool) that have strictly _more_ information than a real commissioner would?  What exactly is a real commissioner supposed to do?